erpnext/selling/report/customer_wise_item_price/customer_wise_item_price.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+
+import frappe
+from erpnext import get_default_company
+from erpnext.accounts.party import get_party_details
+from erpnext.stock.get_item_details import get_price_list_rate_for
+from frappe import _
+
+
+def execute(filters=None):
+	if not filters:
+		filters = {}
+
+	if not filters.get("customer"):
+		frappe.throw(_("Please select a Customer"))
+
+	columns = get_columns(filters)
+	data = get_data(filters)
+
+	return columns, data
+
+
+def get_columns(filters=None):
+	return [
+		{
+			"label": _("Item Code"),
+			"fieldname": "item_code",
+			"fieldtype": "Link",
+			"options": "Item",
+			"width": 150
+		},
+		{
+			"label": _("Item Name"),
+			"fieldname": "item_name",
+			"fieldtype": "Data",
+			"width": 200
+		},
+		{
+			"label": _("Selling Rate"),
+			"fieldname": "selling_rate",
+			"fieldtype": "Currency"
+		},
+		{
+			"label": _("Available Stock"),
+			"fieldname": "available_stock",
+			"fieldtype": "Float",
+			"width": 150
+		},
+		{
+			"label": _("Price List"),
+			"fieldname": "price_list",
+			"fieldtype": "Link",
+			"options": "Price List",
+			"width": 120
+		}
+	]
+
+
+def get_data(filters=None):
+	data = []
+	customer_details = get_customer_details(filters)
+	items = get_selling_items(filters)
+
+	for item in items:
+		price_list_rate = get_price_list_rate_for(customer_details, item.item_code) or 0.0
+		available_stock = frappe.db.sql("SELECT sum(actual_qty) FROM `tabBin` WHERE item_code = %s", item.item_code)
```

Comment:
first, get all items and convert it into the list.
``` items = [d.item_code for d in frappe.get_all("Item", filters=item_filters, fields=["item_code", "item_name"], order_by="item_name")]```

And then you can try

```actual_qty = frappe.get_all("Bin", filters = [('item_code', 'IN', items)], fields = ["actual_qty"])```

After that sum up only two queries is required.
so first, it will bring all data.

Reply:
@Anurag810 You might be looking at an older version of the code. I already fixed this with a single `get_all` query on Bin, and then I process it later.

Reply:
Yeah, my mistake.